code/modules/mob/living/carbon/human/species/unathi.dm

```diff
@@ -128,3 +125,33 @@
 		"appendix" = /obj/item/organ/internal/appendix,
 		"eyes" =     /obj/item/organ/internal/eyes/unathi
 		)
+
+/datum/species/unathi/ashwalker/on_species_gain(mob/living/carbon/human/H)
+	var/datum/action/innate/ignite/ash_walker/fire = new()
+	fire.Grant(H)
+
+/datum/species/unathi/ashwalker/on_species_loss(mob/living/carbon/human/H)
+	..()
+	for(var/datum/action/innate/ignite/ash_walker/fire in H.actions)
+		fire.Remove(H)
+
+/datum/action/innate/ignite/ash_walker
+	desc = "Allows you to light a fire once every five minutes."
+	var/cooldown = 0
```

Comment:
I don't think you decrease this var anywhere? Consider adding a timer whenever you use the ability, then make a new proc that just sets a `on_cooldown` var to false or something. Feel free to ping me on discord if you need help with this, I didn't explain it very well here

Reply:
Hal said it's better to do the way I did it